Dockerfile

```diff
@@ -108,12 +103,23 @@ ENV PATH="/usr/local/cargo/bin:/usr/local/rustup/bin:${PATH}"
 ENV RUSTUP_HOME="/usr/local/rustup"
 
 # Note, we need to mount the .git folder here so that setuptools-scm is able to fetch git tag for version.
+# Includes a hack to install tensorstore 0.1.45, which doesn't distribute a pypi wheel for python 3.12, and the metadata
+# in the source distribution doesn't match the expected pypi version.
 RUN --mount=type=bind,source=./.git,target=./.git \
   --mount=type=bind,source=./requirements-test.txt,target=/requirements-test.txt \
   --mount=type=bind,source=./requirements-cve.txt,target=/requirements-cve.txt \
   <<EOF
 set -eo pipefail
-uv pip install maturin --no-build-isolation && uv pip install --no-build-isolation \
+uv pip install maturin --no-build-isolation --break-system-packages
+
+pip install --use-deprecated=legacy-resolver  --no-build-isolation --break-system-packages \
```

Comment:
can you please add a comment on why we need this, and also add a global comment to the dockerfile as to why `--break-system-packages` is used (I know it's because of the new python venv requirement...but let's just document it to make it easier for everyone.

Reply:
So I added a single `UV_BREAK_SYSTEM_PACKAGES` env variable; looks like the NGC torch image team already set the equivalent pip variable; 
```Dockerfile
ENV PIP_BREAK_SYSTEM_PACKAGES=1
```
https://gitlab-master.nvidia.com/dl/dgx/pytorch/-/blob/main/Dockerfile.base#L43